graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -59,77 +90,80 @@ def rename_schema(ast, renamings):
 
     Args:
-        ast: Document, the schema that we're returning a modified version of
-        renamings: Dict[str, str], mapping original type/interface/enum name to renamed name. If
-                   a name does not appear in the dict, it will be unchanged
-        query_type: str, name of the query type, e.g. 'RootSchemaQuery'
-        scalars: Set[str], the set of all scalars used in the schema, including user defined
-                 scalars and and used builtin scalars, excluding unused builtins
+        schema_ast: schema that we're returning a modified version of
+        renamings: maps original type/interface/enum name to renamed name. Any name not in the dict
+                   will be unchanged
+        query_type: name of the query type, e.g. 'RootSchemaQuery'
+        scalars: set of all scalars used in the schema, including user defined scalars and used
+                 builtin scalars, excluding unused builtins
 
     Returns:
-        Tuple[Document, Dict[str, str]], containing the modified version of the AST, and
-        the renamed type name to original type name map. Map contains all types, including
-        those that were not renamed.
+        Tuple containing the modified version of the schema AST, and the renamed type name to
+        original type name map. Map contains all types, including those that were not renamed.
 
     Raises:
         - InvalidTypeNameError if the schema contains an invalid type name, or if the user attempts
           to rename a type to an invalid name
         - SchemaNameConflictError if the rename causes name conflicts
     """
     visitor = RenameSchemaTypesVisitor(renamings, query_type, scalars)
-    renamed_ast = visit(ast, visitor)
+    renamed_schema_ast = visit(schema_ast, visitor)
 
-    return renamed_ast, visitor.reverse_name_map
+    return renamed_schema_ast, visitor.reverse_name_map
 
 
-def _rename_query_type_fields(ast, renamings, query_type):
+def _rename_query_type_fields(
+    schema_ast: DocumentNode, renamings: Mapping[str, str], query_type: str
+) -> DocumentNode:
     """Rename all fields of the query type.
 
-    The input AST will not be modified.
+    The input schema AST will not be modified.
 
     Args:
-        ast: DocumentNode, the schema that we're returning a modified version of
-        renamings: Dict[str, str], mapping original field name to renamed name. If a name
-                   does not appear in the dict, it will be unchanged
-        query_type: string, name of the query type, e.g. 'RootSchemaQuery'
+        schema_ast: schema that we're returning a modified version of
+        renamings: maps original query type field name to renamed name. Any name not in the dict
+                   will be unchanged
+        query_type: name of the query type, e.g. 'RootSchemaQuery'
 
     Returns:
-        DocumentNode, representing the modified version of the input schema AST
+        Modified version of the input schema AST
```

Comment:
```suggestion
        modified version of the input schema AST
```
Also, is this type guaranteed to be the same as the input type? If so, perhaps another opportunity for a generic?

Reply:
Nothing inherits from DocumentNode so the current annotation effectively does the same thing as a generic here